evm/src/main/java/org/hyperledger/besu/evm/frame/Memory.java

```diff
@@ -192,6 +194,19 @@ int getActiveWords() {
    *     numBytes}.
    */
   public Bytes getBytes(final long location, final long numBytes) {
+    return getBytes(location, numBytes, false);
+  }
+
+  /**
+   * Returns a copy of bytes from memory.
+   *
+   * @param location The location in memory to start with.
+   * @param numBytes The number of bytes to get.
+   * @param shadow If set, perform a “shadow read”, i.e. do not increase the active words.
+   * @return A fresh copy of the bytes from memory starting at {@code location} and extending {@code
+   *     numBytes}.
+   */
+  public Bytes getBytes(final long location, final long numBytes, final boolean shadow) {
```

Comment:
Rather than layering on a "shadow" flag that results two distinct forks of a function, I would rather see a `getBytesWithoutGrowth` that will zero pad without growing the memory size.

Reply:
Fair point, addressed in the latest patch.